src/core/session.py

```diff
@@ -224,12 +225,19 @@ def addGps(
     ):
         self._gpsLayer.add(longitude, latitude, hour, minu, sec)
 
-    def _addFeature(self, layer: QgsVectorLayer, dt: str = "") -> None:
+    def _addFeature(
+        self,
+        layer: QgsVectorLayer,
+        dt: str = "",
+        geom: QgsGeometry = QgsGeometry(),
+    ) -> None:
         feat = QgsVectorLayerUtils.createFeature(layer)
 
         if not dt:
             dt = utils.now()
         feat["dateTime"] = dt
+        if geom:
```

Comment:
Is it enough to detect an empty geometry in PyQGIS?

Reply:
Yes, I checked it :)

```python
>>> bool(QgsGeometry())
False
```

Reply:
ok good to know :+1: